### DISCUSSION TITLE
Is it possible with EDC to have client credentials for Backend Service (receiver.http.endpoint) and Notification Receiver Endpoint

### DISCUSSION THREAD

**Discussion by likhithreddy153215** [eclipse-tractusx/tractusx-edc#567] [Q&A]
Dear Team,

Is it possible with EDC to have client credentials for Backend Service (receiver.http.endpoint) and Notification Receiver Endpoint

Background: As Backend Service and Notification Receiver Endpoint could be in different cloud environments and  EDC in different Cloud environment and hence use  authentification (client credentials) for security reasons. 

so it important that the communication between Backend Service (receiver.http.endpoint) and EDC and EDC and Notification Receiver Endpoint is secure.

1.Backend Service used for receiving Endpoint Data Reference

<img width="371" alt="MicrosoftTeams-image (15)" src="https://github.com/eclipse-tractusx/tractusx-edc/assets/138694069/afd299b3-26bc-4c7d-9085-44d2de0ea1fd">

2.Notification Receiver Endpoint

<img width="385" alt="MicrosoftTeams-image (14)" src="https://github.com/eclipse-tractusx/tractusx-edc/assets/138694069/830b3f67-682c-4d96-a40f-91947849c076">

**Comment by wolf4ood** [comment]
Hi @likhithreddy153215 

For the 1) we are using the EDC extension [here](https://github.com/eclipse-edc/Connector/tree/main/extensions/control-plane/transfer/transfer-pull-http-dynamic-receiver).

You can configure `auth-key` and `auth-code` that would be used in header for authentication when receiving the EDR

**Reply by likhithreddy153215** [reply]
Thanks wolf400d for replying. 

Are you giving the auth-key and auth-code in the Transfer dynamic http receiver call or are you hard coding the auth-key and auth-code in the EDC controlplane Config 

This way 
![image](https://github.com/eclipse-tractusx/tractusx-edc/assets/138694069/edac1174-3263-4852-bb05-70cdeb0c00bc)

or other way as env in EDC controlplane config
![image](https://github.com/eclipse-tractusx/tractusx-edc/assets/138694069/71da567d-779c-4282-9b01-856a62399e9b)

**Reply by domreuter** [reply]
@likhithreddy153215 Right now it is option 2. The url can be set dynamically but is always referring to the static properties from the config for authkey and authcode.

**Comment by lgblaumeiser** [comment]
Closed as stale, please renew item if this is still an issue
